druid/src/scroll_component.rs

```diff
@@ -421,65 +385,57 @@ impl<T: Data, W: Widget<T>> Widget<T> for Scroll<T, W> {
         }
     }
 
-    fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle, data: &T, env: &Env) {
+    /// Checks if the lifecycle event applies to the scroll behavior, uses it and returns true if so
+    ///
+    /// Returns false if the lifecycle event was not used
+    pub fn filter_lifecycle(
```

Comment:
I'm not sure I understand. Why would filtering `Size` be a bad idea? The child is given an infinite space to layout in, with no regard for the viewport size.

We need the ability for the child to know how large the viewport is, and their position in it, but that would be on a subscription basis... we don't want to push that down to every child always.

Reply:
Oops, I still had in mind that `Size` is the window size, but that was changed some while ago.
However, `Size` will be filtered by `WidgetPod` anyways so I still think there should be no filtering for lifecycle events by the scroll component.

Reply:
I think there's two things here:
1) ScrollComponent isn't a Widget, so this isn't literally filtering lifecycle events- it's indicating if the scroll functionality the Component aids in implementing is interested in the event proposed. It's up to the widget using the Component to implement scroll functionality whether this affects event routing or not.
2) I'm not sure the WidgetPod would 'know' to filter this event out - they don't really 'know' if they're being laid out and painted on the window surface, or a scroll surface, or the moon, or what. So I don't think they would filter this, just because they have no reason to understand why they would or wouldn't be interested.

I hope that clarifies what's going on and why I think the filtering is still a good idea. Maybe I'm missing some other reason it's a bad idea!

Reply:
I'm going to be honest when I say that I'm not read up on all the possible lifecycle events so I'm open to making any changes in this area once a consensus is reached

Reply:
`ScrollComponent` no longer filters any lifecycle events